packages/fabric8-ui/src/app/home/home.component.ts

```diff
@@ -1,33 +1,66 @@
 import { Component, OnInit, ViewEncapsulation } from '@angular/core';
-import { User, UserService } from 'ngx-login-client';
-import { first } from 'rxjs/operators';
-import { UserSpacesService } from '../shared/user-spaces.service';
+import { User, UserService, AuthenticationService } from 'ngx-login-client';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ProviderService } from '../shared/account/provider.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   encapsulation: ViewEncapsulation.None,
   selector: 'alm-home',
   templateUrl: './home.component.html',
-  styleUrls: ['./home.component.less'],
-  providers: [UserSpacesService],
 })
 export class HomeComponent implements OnInit {
   loggedInUser: User;
-  spacesCount: number = -1;
+  openShiftLinked: boolean = false;
+  subscriptions: Subscription[] = [];
+  errorConnecting: boolean = false;
+  loading: boolean = true;
 
   constructor(
     private readonly userService: UserService,
-    private readonly userSpacesService: UserSpacesService,
+    private readonly auth: AuthenticationService,
+    private readonly route: ActivatedRoute,
+    private readonly providerService: ProviderService,
+    private readonly router: Router,
   ) {}
 
   ngOnInit() {
     this.loggedInUser = this.userService.currentLoggedInUser;
-    this.userSpacesService
-      .getInvolvedSpacesCount()
-      .pipe(first())
-      .subscribe(
-        (spacesCount: number): void => {
-          this.spacesCount = spacesCount;
-        },
-      );
+    if (this.loggedInUser.attributes && this.loggedInUser.attributes.cluster) {
+      this.linkOpenshiftAccounts();
+    }
+  }
+
+  linkOpenshiftAccounts() {
+    this.loading = true;
+    let connectionSub = this.auth
+      .isOpenShiftConnected(this.loggedInUser.attributes.cluster)
+      .subscribe((isConnected) => {
+        this.openShiftLinked = isConnected;
+        let wait = this.route.snapshot.queryParams['wait'];
+        if (!isConnected && !wait) {
+          // first time through and user isn't connected - automatically connect accounts
+          this.providerService.linkOpenShift(
+            this.loggedInUser.attributes.cluster,
+            window.location.origin + '/_home?wait=true',
+          );
+        } else if (!isConnected && wait) {
+          // second time through - do not try again if wait is on URL and the user is still not connected
+          // something happened, show error
+          this.errorConnecting = true;
```

Comment:
@karthikjeeyar @rohitkrai03 
Are we redirecting to an error page? Or are showing a notification? 
What is the message we are showing in case of 'user still not connected'

Reply:
@sbose78 If the user is not connected to openshift, home page will try to connect to openshift in the background. If still some error happens then this screen will be shown to the user - 

<img width="1680" alt="screenshot 2019-01-09 at 7 52 29 pm" src="https://user-images.githubusercontent.com/6041994/50905502-fe68e980-1448-11e9-9d84-e39a6e9f5b90.png">

Reply:
Perfect, thanks.